eva/udfs/decorators/udf_decorators.py

```diff
@@ -0,0 +1,66 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List
+
+from eva.io_descriptors.eva_arguments import IOArgument
+
+
+# decorator for the setup function. It will be used to set the cache, batching and
+# udf_type parameters in the catalog
+def setup(use_cache: bool, udf_type: str, batch: bool):
+    def inner_fn(arg_fn):
+        def wrapper(*args, **kwargs):
+            # TODO set the batch and caching parameters. update in catalog
```

Comment:
This will happen inside the create udf executor. All we need from here is the tags to be updated.